packages/flutter/lib/src/gestures/monodrag.dart

```diff
@@ -384,13 +395,192 @@ abstract class DragGestureRecognizer extends OneSequenceGestureRecognizer {
-        result = _acceptedActivePointers.length <= 1 || pointer == _acceptedActivePointers.last;
+        result = _activePointer == null || pointer == _activePointer;
     }
     return result;
   }
 
+  void _resetMoveDeltaRecords() {
+    if (_frameTimeStamp != null) {
+      _moveDeltaBeforeFrame.clear();
+      _frameTimeStamp = null;
+    }
+
+    assert(_frameTimeStamp == null);
+    assert(_moveDeltaBeforeFrame.isEmpty);
+  }
+
+  void _recordMoveDeltaForMultitouch(int pointer, Offset localDelta) {
+    if (multitouchDragStrategy != MultitouchDragStrategy.maxAllPointers) {
+      assert(_frameTimeStamp == null);
+      assert(_moveDeltaBeforeFrame.isEmpty);
+      return;
+    }
+
+    assert(_frameTimeStamp == SchedulerBinding.instance.currentSystemFrameTimeStamp);
+
+    if (localDelta == Offset.zero) {
+      return;
+    }
+
+    if (_moveDeltaBeforeFrame.containsKey(pointer)) {
+      _moveDeltaBeforeFrame[pointer]!.add(localDelta);
+    } else {
+      _moveDeltaBeforeFrame[pointer] = <Offset>{ localDelta };
+    }
+  }
+
+  double _getSumDelta({
+    required int pointer,
+    required bool positive,
+    required _DragDirection axis,
+  }) {
+    double sum = 0.0;
+
+    if (!_moveDeltaBeforeFrame.containsKey(pointer)) {
+      return sum;
+    }
+
+    for (final Offset delta in _moveDeltaBeforeFrame[pointer]!) {
+      if (positive) {
+        if (axis == _DragDirection.vertical) {
+          if (delta.dy > 0.0) {
+            sum += delta.dy;
+          }
+        } else {
+          if (delta.dx > 0.0) {
+            sum += delta.dx;
+          }
+        }
+      } else {
+        if (axis == _DragDirection.vertical) {
+          if (delta.dy < 0.0) {
+            sum += delta.dy;
+          }
+        } else {
+          if (delta.dx < 0.0) {
+            sum += delta.dx;
+          }
+        }
+      }
+    }
+    return sum;
+  }
+
+  int? _getMaxSumDeltaPointer({
+    required bool positive,
+    required _DragDirection axis,
+  }) {
+    if (_moveDeltaBeforeFrame.isEmpty) {
+      return null;
+    }
+
+    int? ret;
+    double? max;
+    double sum;
+    for (final int pointer in _moveDeltaBeforeFrame.keys) {
+      sum = _getSumDelta(pointer: pointer, positive: positive, axis: axis);
+      if (ret == null) {
+        ret = pointer;
+        max = sum;
+      } else {
+        if (positive) {
+          if (sum > max!) {
+            ret = pointer;
+            max = sum;
+          }
+        } else {
+          if (sum < max!) {
+            ret = pointer;
+            max = sum;
+          }
+        }
+      }
+    }
+    assert(ret != null);
+    return ret;
+  }
+
+  Offset _resolveLocalDeltaForMultitouch(int pointer, Offset localDelta) {
+    if (multitouchDragStrategy != MultitouchDragStrategy.maxAllPointers) {
+      _resetMoveDeltaRecords();
+      return localDelta;
+    }
+
+    final Duration currentSystemFrameTimeStamp = SchedulerBinding.instance.currentSystemFrameTimeStamp;
```

Comment:
Good idea, I'll try it

Reply:
![image](https://github.com/flutter/flutter/assets/61075224/e12c6212-fbf9-4aef-94db-73a51e404d72)
The event timestamp may changes during a frame.

`What happens if the screen is completely static (but there's still a gesture detector working, probably because the gesture's slop isn't reached yet).`

If the gesture is not accepted, the resolved delta will not applied.